lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -518,13 +624,6 @@ def store_new_tls_resources(self, cert_type: CertType, secrets: Dict[str, Any]):
             logging.error("TLS key not found, quitting.")
             return
 
-        # we store the pem format to make it easier for the python requests lib
-        if cert_type == CertType.APP_ADMIN:
```

Comment:
Would we not save a potential deferral in the current unit if we always saved the admin certificate if available?

Reply:
No, this causes trouble in the communication with Opensearch. If we already have the new pem-file but the new `unit-http` cert was not loaded yet, verification of http-requests against any API will fail.

Therefore we update the pem-file only once the certs have been updated (in the `on_tls_conf_set` method).